platform/openide.util.ui/src/org/openide/util/ImageUtilities.java

```diff
@@ -117,18 +124,27 @@
      * {@link Icon#paintIcon(Component, Graphics, int, int)} when converting an {@code Icon} to an
      * {@code Image}. See comment in {@link #icon2ToolTipImage(Icon, URL)}.
      */
-    private static volatile Component dummyIconComponent;
+    private static volatile Component dummyIconComponentLabel;
+
+    /**
+     * Second dummy component. Some {@link Icon#paintIcon(java.awt.Component, java.awt.Graphics, int, int)} are very picky and downcast the
+     * Component to a specific subclass. JCheckBox will satisfy checkboxes, abstract buttons etc. Will not eliminate all cases, but helps.
+     * 
+     */
+    private static volatile Component dummyIconComponentButton;
 
     static {
         /* Could have used Mutex.EVENT.writeAccess here, but it doesn't seem to be available during
         testing. */
         if (EventQueue.isDispatchThread()) {
-            dummyIconComponent = new JLabel();
+            dummyIconComponentLabel = new JLabel();
+            dummyIconComponentButton = new JCheckBox();
```

Comment:
If we're doing a JCheckBox here instead of a JButton, the variable should be called dummyIconComponentCheckBox.

But why are we doing a JCheckBox again? I know it extends JButton but unless you know of specific cases that need it, I'd prefer to just have a JButton. Otherwise the hacks start getting too specific, and we'll be afraid to change them in the future.

Reply:
During development of the UIDefaults replacing code, I used `ImageUtilities` on real icons served by UIManager (default LaF). Several of them had special implementations that downcast the Component to JCheckbox in their `paint()`. Having a JCheckbox satisfies more icons than just JButton (I tried to choose the class that covered most of the errors).

Reply:
OK, as long as you saw actual cases, that's fine. What do you mean by "using ImageUtilities"; which method(s) specifically? And which LAF was this? (Which OS?) Why haven't these cases come up before?

Reply:
> which method(s) specifically
`ImageUtilities.icon2Image`; that ultimately calls `paintIcon()` to get the icon rendered to a `BufferedImage`. OS Linux / KDE, JDK 8+11, default L&F. 

> Why haven't these cases come up before?
Nobody called `icon2Image` on that particular UIManager icon ?

Reply:
I think the reasoning is captured fine in the comment above too.

Reply:
Yeah, it's fine with me. Just wondering if these were problems caused due to aggressive manual testing only, or whether they actually occur during real use.